BaragonService/src/main/java/com/hubspot/baragon/service/managers/ElbManager.java

```diff
@@ -182,11 +195,134 @@ public void syncAll() {
     }
   }
 
+  private void syncListeners(List<LoadBalancerDescription> elbsForGroup, BaragonGroup group) {
+    Collection<BaragonAgentMetadata> groupAgents = loadBalancerDatastore.getAgentMetadata(group.getName());
+    Map<String, Pair<Integer, Integer>> listenersToKeep = new HashMap<String, Pair<Integer, Integer>>();
+
+    // Handle create requests
+    List<CreateLoadBalancerListenersRequest> loadBalancerListenersCreateRequests = new ArrayList<>();
+    for (BaragonAgentMetadata agent : groupAgents) {
```

Comment:
This outer loop is not necessary. Since we can only have a single port per group due to the limitations of the ELB anyways, we should find that port in a separate method then move on to loop through the sources. That method should also validate that the port for all agents present is the same since different ports would be an invalid state